### PR TITLE
Force Merge should reject requests with `only_expunge_deletes` and `max_num_segments` set

### LINKED ISSUE #43102: Force merge with only_expunge_deletes does not honour max_num_segments
Force merge requests where `only_expunge_deletes` is set to `true` do not honour other request parameters such as `max_num_segments`. We should throw an exception when those two parameters are set at the same time, given that we can't do both.

### PR DESCRIPTION
**Pull request by tlrx**
This pull request changes the `ForceMergeRequest.validate()` method so that it does not accept the parameters `only_expunge_deletes` and `max_num_segments` to be set at the same time. 

The motivation is that `InternalEngine.forceMerge()` just ignores the max. number of segments parameter when the only expunge parameter is set to true, leaving the wrong impression to the user that max. number of segments has been applied. This PR also changes `InternalEngine.forceMerge()` so that it now throws an exception when both parameters are set, and modifies tests where needed.

Because it changes the behavior of the REST API I marked this as `>breaking`. Once this is merged I'll open a follow up PR to add deprecation logging in 7.x.

Closes  #43102

### CODE REVIEW

**Comment by tlrx** [comment]
@elasticmachine update branch

**Comment by tlrx** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

File: docs/reference/migration/migrate_8_0/indices.asciidoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[float]
+[[breaking_80_indices_changes]]
+=== Force Merge API changes
```
Comment (dnhatn) [inline docs/reference/migration/migrate_8_0/indices.asciidoc]: ++

**Review by dnhatn — APPROVED** [review]
LGTM

**Comment by tlrx** [comment]
Thanks @dnhatn !

**Pull request merged**